builder/parser/line_parsers.go

```diff
@@ -209,34 +208,28 @@ func parseString(rest string) (*Node, map[string]bool, error) {
 
 // parseJSON converts JSON arrays to an AST.
 func parseJSON(rest string) (*Node, map[string]bool, error) {
-	var (
-		myJson   []interface{}
-		next     = &Node{}
-		orignext = next
-		prevnode = next
-	)
-
+	var myJson []interface{}
 	if err := json.Unmarshal([]byte(rest), &myJson); err != nil {
 		return nil, nil, err
 	}
 
+	var top, prev *Node
 	for _, str := range myJson {
 		switch str.(type) {
```

Comment:
Do we need a type switch rather than a single type assertion? I don't think it changes anything besides readability, but I'd find `if s, ok := str.(string); !ok {` easier (and avoids casting twice).